software/firmware/europi.py

```diff
@@ -622,6 +622,64 @@ def value(self, value):
             self.off()
 
 
+class Thermometer:
+    """
+    Wrapper for the temperature sensor connected to Pin 4
+    Reports the module's current temperature in Celsius.
+    If the module's temperature sensor is not working correctly, the temperature will always be reported as None
+    """
+
+    # Conversion factor for converting from the raw ADC reading to sensible units
+    # See Raspberry Pi Pico datasheet for details
+    TEMP_CONV_FACTOR = 3.3 / 65535
+
+    def __init__(self):
+        # The Raspberry Pi Pico 2's temperature sensor doesn't with some older
+        # uPython firmwares so do some basic exception handling
+        try:
+            self.pin = ADC(PIN_TEMPERATURE)
+        except:
+            self.pin = None
+
+    def read_temperature(self):
+        """
+        Read the ADC and return the current temperature
+        @return  The current temperature in Celsius, or None if the hardware did not initialze properly
+        """
+        if self.pin:
+            # see the pico's datasheet for the details of this calculation

```

Comment:
[Spelling]
Capital letter for See and Pico

Reply:
Done, though this comment was copied verbatim from the diagnostic script. How picky are we in general about punctuation and caps in explanatory comments?